blt/src/Blt/Plugin/Commands/AcquiaApi.php

```diff
@@ -0,0 +1,239 @@
+      ],
+    ];
+    return $this->callAcquiaApi("/environments/{$this->envIds[$environment]}/ssl/certificates", 'POST', $data);
+  }
+
+  /**
+   * Activate an SSL cert already installed on the environment.
+   *
+   * @param string $environment
+   *   Environment to effect.
+   * @param int $certId
+   *   Certificate ID to activate.
+   *
+   * @return bool|string
+   *   API Response.
+   */
+  public function activateCert($environment, $certId) {
+    return $this->callAcquiaApi("/environments/{$this->envIds[$environment]}/ssl/certificates/{$certId}/actions/activate", 'POST');
+  }
+
+  /**
+   * Remove a cert from the environment.
+   *
+   * @param string $environment
+   *   Environment to effect.
+   * @param int $certId
+   *   Certificate ID to remove.
+   *
+   * @return bool|string
+   *   API Response.
+   */
+  public function removeCert($environment, $certId) {
+    return $this->callAcquiaApi("/environments/{$this->envIds[$environment]}/ssl/certificates/{$certId}", 'DELETE');
+  }
+
+  /**
+   * Get all SSL certs that are on the environment.
+   *
+   * @param string $environment
+   *   Environment to effect.
+   *
+   * @return bool|array
+   *   API Response.
+   */
+  public function getCerts($environment) {
+    if ($response = $this->callAcquiaApi("/environments/{$this->envIds[$environment]}/ssl/certificates")) {
+      return json_decode($response, TRUE);
+    }
+    return $response;
+  }
+
+  /**
+   * Deploy a git branch or tag to a certain environment.
+   *
+   * @param string $environment
+   *   Environment to effect.
+   * @param string $reference
+   *   Git branch or tag name.
+   *
+   * @return bool|string
+   *   API Response.
+   */
+  public function deployCode($environment, $reference) {
+    return $this->callAcquiaApi("/environments/{$this->envIds[$environment]}/code/actions/switch", 'POST', ['json' => ['name' => $reference]]);
+  }
+
+  /**
+   * Make an API call to Acquia Cloud API V2.
+   *
+   * @param string $path
+   *   API Endpoint, options from: https://cloudapi-docs.acquia.com/.
+   * @param string $method
+   *   Request method: GET, POST, PUT, DELETE.
+   * @param array $options
+   *   Request options for post json data or headers.
+   *
+   * @return bool|string
+   *   False if it fails, api response string if success.
+   *
+   * @see https://docs.acquia.com/acquia-cloud/develop/api/auth/
+   */
+  protected function callAcquiaApi($path, $method = 'GET', array $options = []) {
+    try {
+      $provider = new GenericProvider([
+        'clientId' => $this->key,
+        'clientSecret' => $this->secret,
+        'urlAuthorize' => '',
+        'urlAccessToken' => 'https://accounts.acquia.com/api/auth/oauth/token',
+        'urlResourceOwnerDetails' => '',
+      ]);
+
+      // Try to get an access token using the client credentials grant.
+      $accessToken = $provider->getAccessToken('client_credentials');
+    }
+    catch (\Exception $e) {
+      echo $e->getMessage();
```

Comment:
Echo? Why not use `$this->say()`?

Reply:
because this is a bare bones class without all the robo/blt stuff. I wasn't quite sure what might be the best for this

Reply:
Oh, I see you're not extending anything. Why not extend the Robo tasks?

Reply:
i suppose extending robo would be better than extending blt. that sounds good

Reply:
oh now i remember. by doing `say()` it doesn't actually output the response. thats why i used echo. I dont know why it doesnt

Reply:
The output should probably use the api as robo has a `stop-on-error` option. Perhaps `printError()` https://github.com/consolidation/Robo/blob/1.x/src/Log/ResultPrinter.php#L64